code/game/machinery/engi_points_vendor.dm

```diff
@@ -0,0 +1,197 @@
+						spawn_atom_to_turf(/obj/item/clothing/head/helmet/space/chronos, D, prize.amount, admin_spawn=FALSE)
+					if(prize.cost >= 1000)
+						radio.talk_into(src, "[usr] has bought [prize.equipment_name] for [prize.cost] points")
+					feedback_add_details("Engi_equipment_bought","[src.type]|[prize.equipment_path]")
+	updateUsrDialog()
+
+/obj/machinery/engi_points_manager/attackby(obj/item/I, mob/user, params)
+	return ..()
+
+/obj/machinery/engi_points_manager/process()
+	power_export_bonus = 0
+	for(var/obj/machinery/power/exporter/PE in power_exporter_list)
+		power_export_bonus = PE.drain_rate/200 // basically controls the balance of the current point system
+	if(GBP_alarm_cooldown <= world.time)
+		for(var/obj/machinery/computer/station_alert/SA in machines)
+			if(SA.z == src.z)
+				air_alarm_bonus = max(0,(1000 - (SA.air_alarm_count * 200)))
+				power_alarm_bonus = max(0,(1000 - (SA.power_alarm_count * 200)))
+				fire_alarm_bonus = max(0,(500 - (SA.fire_alarm_count * 100)))
+				total_bonus = air_alarm_bonus + power_alarm_bonus + fire_alarm_bonus
+				break
+		switch(total_bonus)
+			if(0)
+				alarm_rating = "GREYTIDE IN YELLOW JUMPSUITS"
+				playsound(src, 'sound/misc/compiler-failure.ogg', 100, 1)
+			if(100 to 900)
+				alarm_rating = "COMPLICIT IN THE STATION'S DOWNFALL"
+				playsound(src, 'sound/misc/compiler-failure.ogg', 100, 1)
+			if(1000 to 1500)
+				alarm_rating = "HALF-ASSED"
+				playsound(src, 'sound/misc/compiler-stage1.ogg', 100, 1)
+			if(1600 to 2000)
+				alarm_rating = "ADEQUATE AND UNREMARKABLE"
+				playsound(src, 'sound/misc/compiler-stage1.ogg', 100, 1)
+			if(2100 to 2400)
+				alarm_rating = "IMPRESSIVE"
+				playsound(src, 'sound/misc/compiler-stage2.ogg', 100, 1)
+			if(2500 to 9999999)
+				alarm_rating = "ABSOLUTELY FLAWLESS"
+				playsound(src, 'sound/misc/compiler-stage2.ogg', 100, 1)
+		radio.talk_into(src,"UPDATE: The engineering department has been awarded [air_alarm_bonus] points for the state of the station's air, [power_alarm_bonus] points for the state of the station's power, and [fire_alarm_bonus] points for the state of the station's fire alarms.")
+		radio.talk_into(src,"This bonus represents [((total_bonus)/2500)*100]% of the total possible bonus. Your rating is: [alarm_rating]. Consult the station alert console for details.")
+		if((total_bonus - prior_bonus) >= 1600)
+			radio.talk_into(src,"Congratulations! Due to the significant repairs made by the engineering team, your bonus has been doubled this cycle!")
+			total_bonus = total_bonus*2
+		prior_bonus = total_bonus
+		GBP_alarm_cooldown = world.time + 3000
+		power_export_bonus += (air_alarm_bonus + power_alarm_bonus + fire_alarm_bonus)
+	GBP += power_export_bonus
+	GBPearned += power_export_bonus
+
+/obj/machinery/engi_points_delivery
+	name = "Engineering Reward Fabricator"
+	desc = "Like a christmas tree for engineers"
+	icon = 'icons/obj/machines/engi_points.dmi'
+	icon_state = "geardist"
+	density = 1

```

Comment:
@Robustin it makes it easier for future developers reading your code to make mental assumptions about the type of a variable and the values it will hold.